server.py

```diff
@@ -23,53 +28,137 @@
 def hello():
     return 'hello!'
 
-# Endpoint for PUT method
-@app.route('/write', methods=['PUT'])
+@app.route('/redirect')
+def redirect_endpoint():
+    return redirect('http://localhost:5000/read?uuid=xxx', code=302)
+
+# Endpoint for write method
+@app.route('/write', methods=['POST'])
 def write_file():
     file = request.files['file']
     if file:
         filename = secure_filename(file.filename)
         file_uuid = str(uuid.uuid4())
-        # return str(os.path.join(app.config['UPLOAD_FOLDER'], file_uuid)), 200
         file.save(os.path.join(app.config['UPLOAD_FOLDER'], file_uuid))
+        metadata = metadata_manager.MetadataManager()
+        metadata.update_file_stored(file_uuid, app.config['HOST'])
+        metadata.close_connection()
         return file_uuid, 201
     return 'Write Failed', 500
 
-# Endpoint for GET method
+# Endpoint for read method
 @app.route('/read', methods=['GET'])
 def read_file():
-    sys.stdout.flush()
     filename = request.args.get('uuid')
-    return send_from_directory(UPLOAD_FOLDER, secure_filename(filename))
+    file_path = UPLOAD_FOLDER + '/' + secure_filename(filename)
+    metadata = metadata_manager.MetadataManager()
+    if (os.path.exists(file_path)):
```

Comment:
This is perhaps not a big deal, but I'm wondering whether this is potentially expensive (I/O vs. DB query) - isn't it easier just to call DB and check if it exists locally first before doing this?